packages/ui/src/components/va-data-table/VaDataTable.new.demo.vue

```diff
@@ -0,0 +1,321 @@
+<template>
+  <VbDemo>
+    <VbCard title="Reactive data">
+      <button @click="shuffleItems">Shuffle</button>
+      <va-data-table :items="items" />
+    </VbCard>
+
+    <VbCard title="Use custom slots for address and company">
+      <va-data-table :items="items">
+        <template #head(address)>Street</template>
+        <template #head(company)>Company Name</template>
+
+        <template #cell(address)="{ source: address }">{{ address.street }}</template>
+        <template #cell(company)="{ source:company }">{{ company.name }}</template>
+      </va-data-table>
+    </VbCard>
+
+    <VbCard title="Show footer and append static rows everywhere">
+      <va-data-table :items="items" foot-clone>
+        <template #headAppend>
+          <tr>
+            <th colspan="4">User info</th>
+            <th colspan="4">Contact info</th>
+          </tr>
+        </template>
+
+        <template #bodyAppend>
+          <tr>
+            <td colspan="8">Custom cell which span 8 cells</td>
+          </tr>
+        </template>
+
+        <template #footAppend>
+          <tr>
+            <th colspan="8">Span 8 cells</th>
+          </tr>
+        </template>
+      </va-data-table>
+    </VbCard>
+
+    <VbCard title="Use `colgroup` slot to set 2nd column's width to 50px">
+      <va-data-table :items="items">
+        <template #colgroup>
+          <col>
+          <col width="50">
+        </template>
+      </va-data-table>
+    </VbCard>
+
+    <VbCard title="Filtering">
+      <p>Number of filtered items: {{ filteredCount }}</p>
+      <input type="text" v-model="filter">
+
+      <input id="ucf" type="checkbox" v-model="useCustomFilteringFn">
+      <label for="ucf">Use custom filtering function (looks for an exact match)</label>
+
+      <va-data-table
+        :items="items"
+        :filter="filter"
```

Comment:
I would suggest `filter-text` and `filter` (as function), instead of current naming.
Checking other solutions would make sense though. But `filter` implies for me we're dealing with function.